src/coreclr/jit/scev.cpp

```diff
@@ -982,19 +1019,92 @@ Scev* ScalarEvolutionContext::Simplify(Scev* scev)
                 ScevConstant* cns1 = (ScevConstant*)op1;
                 ScevConstant* cns2 = (ScevConstant*)op2;
                 int64_t       newValue;
-                if (binop->TypeIs(TYP_INT))
+                if (genTypeSize(binop->Type) == 4)
                 {
-                    newValue = FoldBinop<int32_t>(binop->Oper, static_cast<int32_t>(cns1->Value),
-                                                  static_cast<int32_t>(cns2->Value));
+                    newValue = FoldBinop<uint32_t>(binop->Oper, static_cast<uint32_t>(cns1->Value),
+                                                   static_cast<uint32_t>(cns2->Value));
```

Comment:
Changed this since signed overflow is undefined behavior while we want to have wraparound behavior.